### ISSUE TITLE
Is it possible to manually select all tables and mappings and add more than 1 generator to 1 proj?

### ISSUE THREAD

**Issue by CorsairRO** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#433]
Hello,

First of all I would like to thank you very much for the big effort that you made with the tool , which is an amazing tool for me as I could read doc and watch pluralgisht course.

I am very new to EF Core, and I have to move fast a legacy application using EF 5 to EF Core 2.
This is how I come to the reverse code first generator page, as EF Core has no EDMX.

Because I don't find any answer for my questions even in the pluralsight course that I bought and watched, I would kindly ask you 2 questions.

My questions are not touching at all migrations as I just found then unusable from the beginning of them in app with 500 tables and 1000 stored procedures that does lots of things and are very interconnected with each other, and with dba's in the team to handle this.

1) Because my context will have all 500 tables like I said, I would like to have 1 ctx per app module.
Now, I don't want to add 50 dll projects to have 1 generator in each. So my question is if there is supported to have multiple generators in the same project, in different folders, let's say per module

2) I can understand that it is possible for example to create 2 folders for POCO's, for example, 1 for Orders, one for Customers, even if the answer to 1) will be no, It will still be possible in 2 dll's
Now, my question is, let's say both of the POCO FOlders share some object / table, for example Address

I saw and understood that there is a callback to change table name mapping, which is very nice especially for legacy databases, but I see no possibility of changing the namespaces of individual table/poco, like I will explain in next lines

In OrdersContext will be a DbSet for Address, and in CustomersContext the same.

The desired folder structure will be:
DataProject
- Ordering
- - Orders - OrdersContext + all POCOs
- - Customers - CustomersContext + all POCOs
- Address ( only 1 POCO )

Can I somehow move the Address to an upper level ( folder ) , like Adding the POCO for Address to Ordering folder in the upper example, and the DBSets from Customer and Order to share the object between 2 contexts?

Like OrdersContext for example to generate to contain a DBSet<DataProject.Ordering.Address> instead of the DBSet<DataProject.Ordering.Orders.Address> and then add another one POCO to Customers POCO's ?

The reasons for this need are coming from 
- Our DBA's or even ourselves the programmers frequently add/change columns in DB
Now, if  column changes in a table for example from int to string, it will be more convenient for tables used in 20 modules to be regenerated from DB only once, and not regenerate all 20 times for each instance of generator that uses it.
- I have metadata classes with data annotations validators that validates directly the POCO objects.
I have tested this way to work in applications for more than 7 years now, and it works like a charm and I don't want to duplicate the validation codes there.

Again, I underline that I don't care about migrations at all, so if any solution, even manual work for me or my team would be highly appreciated

Thank you very much for reading this, and for your qualified answers and also for the great tool that you are developing there
Have a nice day there. Also if I can help with any contribution implementation there I would be very glad to do so

**Comment by sjh37** [comment]
Hi Corsair,

My company is the same as yours, lots of tables 1000+ and lots of stored procedures 600+, and lots of apps 500+.
We started off with the design like yours, a single data DLL packaged into Nuget, and all the other apps used the nuget package.
Good = A single POCO class for address/order/etc could be passed to any business logic application to be validated because they reference the same type.
Bad = The database context was too large as it contained too many tables. Every time an application started to process some data, it would take over 60 seconds just for the EF database context to validate it's model before anything could be done. SaveChanges() were very very slow because EF had to check every table in db context for modifications so it could generate SQL.

Therefore we moved away from having a single module which contained the whole database.

Instead, each module has its generator, and using filtering to limit the tables/stored procedures to just the ones needed for the module.
Good = The database context is small, and it takes < 1 second for the application to start. SaveChanges() is also fast as it only has to check a few tables.
Bad = No sharing of objects. However we overcame this by calling common modules and passing in the correct poco by using AutoMapper to do the conversion.

We are much happier having multiple small db contexts than one giant db context. Save yourself the pain we went through, and go for lots of small db contexts.
